pkg/rbd/rbd_util.go

```diff
@@ -558,6 +595,21 @@ func hasSnapshotFeature(imageFeatures string) bool {
 	return false
 }
 
+// imageFeaturesToUint takes the comma separated image features and converts
+// them to a RbdImageOptionFeatures value.
+func imageFeaturesToUint(ctx context.Context, imageFeatures string) uint64 {
+	features := uint64(0)
+
+	for _, f := range strings.Split(imageFeatures, ",") {
```

Comment:
This logic basically means, we only account `layering` feature.  Cant we just check for `,layering,` in the string (.Contains) and return `uint64` value ? why because, as this func returns unique/single `uint64` value, I doubt this looping and condition check going to make sense even in future for more options.

Reply:
The next release of go-ceph will have a way to map feature strings to uint64's in a better way. When that is available, this function will be rewritten so that new features automatically will be supported.

See https://github.com/ceph/go-ceph/blob/master/rbd/features.go#L110 for more details.